hr_leave_dashboard_extra_hours/models/hr_leave_type.py

```diff
@@ -0,0 +1,170 @@
+                "hr.employee"
+            ]._get_contextual_employee()
+            extra_hours_data = extra_hours_time_off_type._prepare_extra_hours_data(
+                employee, target_date
+            )
+            allocation_data.append(extra_hours_data)
+        return allocation_data
+
+    def _prepare_lt_info_detail(self):
+        return {
+            "remaining_leaves": 0,
+            "virtual_remaining_leaves": 0,
+            "max_leaves": 0,
+            "accrual_bonus": 0,
+            "leaves_taken": 0,
+            "virtual_leaves_taken": 0,
+            "leaves_requested": 0,
+            "leaves_approved": 0,
+            "closest_allocation_remaining": 0,
+            "closest_allocation_expire": False,
+            "holds_changes": False,
+            "total_virtual_excess": 0,
+            "virtual_excess_data": {},
+            "exceeding_duration": 0,
+            "request_unit": self.request_unit,
+            "icon": self.sudo().icon_id.url,
+            "allows_negative": self.allows_negative,
+            "max_allowed_negative": self.max_allowed_negative,
+        }
+
+    def _prepare_extra_hours_data(self, employee, target_date):
+        if target_date and isinstance(target_date, str):
+            target_date = datetime.fromisoformat(target_date).date()
+        elif target_date and isinstance(target_date, datetime):
+            target_date = target_date.date()
+        elif not target_date:
+            target_date = fields.Date.today()
+        allocations_leaves_consumed, extra_data = employee.with_context(
+            ignored_leave_ids=self.env.context.get("ignored_leave_ids")
+        )._get_consumed_leaves(self, target_date)
+        lt_info = self._prepare_lt_info_detail()
+        lt_info["exceeding_duration"] = extra_data[employee][self]["exceeding_duration"]
+        for excess_date, excess_days in extra_data[employee][self][
+            "excess_days"
+        ].items():
+            amount = excess_days["amount"]
+            lt_info["virtual_excess_data"][
+                excess_date.strftime("%Y-%m-%d")
+            ] = excess_days
+            lt_info["total_virtual_excess"] += amount
+            if not self.allows_negative:
+                continue
+            lt_info["virtual_leaves_taken"] += amount
+            lt_info["virtual_remaining_leaves"] -= amount
+            if excess_days["is_virtual"]:
+                lt_info["leaves_requested"] += amount
+            else:
+                lt_info["leaves_approved"] += amount
+                lt_info["leaves_taken"] += amount
+                lt_info["remaining_leaves"] -= amount
+        allocations_now = self.env["hr.leave.allocation"]
+        allocations_date = self.env["hr.leave.allocation"]
+        allocations_with_remaining_leaves = self.env["hr.leave.allocation"]
+        for allocation, data in allocations_leaves_consumed[employee][self].items():
+            if allocation:
+                today = fields.Date.today()
```

Comment:
today date can go out of the loop

Reply:
Hi @santostelmo and @leemannd, I have added some unit tests.